libs/langchain-openai/src/azure/chat_models.ts

```diff
@@ -541,6 +547,44 @@ export class AzureChatOpenAI extends ChatOpenAI {
       delete json.kwargs.azure_openai_api_key;
       delete json.kwargs.azure_openai_api_version;
       delete json.kwargs.azure_open_ai_base_path;
+
+      if (!json.kwargs.azure_endpoint && this.azureOpenAIEndpoint) {
+        json.kwargs.azure_endpoint = this.azureOpenAIEndpoint;
+      }
+      if (!json.kwargs.azure_endpoint && this.azureOpenAIBasePath) {
+        const parts = this.azureOpenAIBasePath.split("/openai/deployments/");
+        if (parts.length === 2 && parts[0].startsWith("http")) {
+          const [endpoint] = parts;
+          json.kwargs.azure_endpoint = endpoint;
+        }
+      }
+      if (!json.kwargs.azure_endpoint && this.azureOpenAIApiInstanceName) {
+        json.kwargs.azure_endpoint = `https://${this.azureOpenAIApiInstanceName}.openai.azure.com/`;
+      }
```

Comment:
I would wrap all these in an if statement:

```typescript
if (!json.kwargs.azure_endpoint) {
  if (this.azureOpenAIEndpoint) {
    ...
  } else if (this.azureOpenAIBasePath) {
    ...
  } else if (this.azureOpenAIApiInstanceName) {
    ...
  }
}
```

same with the `!json.kwargs.deployment_name` statements below